src/metadata.rs

```diff
@@ -953,15 +956,15 @@ impl TimestampMetadataBuilder {
 pub struct TimestampMetadata {
     version: u32,
     expires: DateTime<Utc>,
-    snapshot: MetadataDescription,
+    meta: HashMap<MetadataPath, MetadataDescription>,
```

Comment:
By implementing it this way, you accept timestamp.json files which contain roles other than snapshot.json to be valid. I can't think of why you'd want to put another role in the timestamp, and the workflow doesn't use that information. Would it make sense to only allow snapshot in timestamp?

The TUF spec isn't really clear on this. What do python-tuf and go-tuf do?

Reply:
Well, technically, I think you can list anything you like in the timestamp other than metadata about snapshot, so long as you ignore them, although it should be a warning flag

So it's safe to say that for all practical purposes, anything other than metadata about snapshot in the timestamp metadata should be rejected

What do you think, @JustinCappos?